### PR TITLE
Add introduction of High  Available

### PR DESCRIPTION
**Pull request by sunzhaoyang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-ha.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+title: Data Migration 高可用机制
+---
+
+# Data Migration 高可用机制
+
+为了保障迁移服务的连续性，DM 中的组件 DM Master 和 DM Worker 均实现了高可用机制，避免单点故障。本文将详述其高可用机制的内部机制，以及对迁移任务的影响。
+
+在详细了解之前，首先要了解 DM 整体架构以及各个关键组件的作用，请参考 [DM 架构](/dm/dm-arch.md)。
```
Comment (ran-huang) [inline dm/dm-ha.md:9]: [DM 架构](https://github.com/pingcap/docs-cn/blob/master/dm/dm-arch.md#%E9%AB%98%E5%8F%AF%E7%94%A8)里有讲高可用的一节，请整合两部分内容。

btw 现在这篇感觉过于详细了，用户手册里需要深入到这种程度吗？放到 internal 论坛/源码阅读里是不是更合适...

File: dm/dm-ha.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+## DM Master 内部组成
+
+![dm-ha-1](/media/dm/dm-ha-1.png)
+
+- **Grpc 和 Http 接口。** DM Master 对外提供 Grpc 以及 Http 接口给其他组件，如 dmctl，WebUI，DM worker调用，DM Master 的 Follower 节点，收到 Grpc 和 Http 请求也会 redirect 给 DM Master 的 Leader 节点进行处理。
```
Comment (ran-huang) [inline dm/dm-ha.md:15]: 全文替换：

- Grpc -> gRPC
- Http -> HTTP
- Dmctl -> dmctl
- ETCD -> etcd

File: dm/dm-high-availability.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+为了保障迁移服务的连续性，DM 中的组件 DM master 和 DM worker 均实现了高可用机制，避免单点故障。本文将详述其高可用机制的内部机制，以及对迁移任务的影响。
+
+在详细了解 DM 高可用机制之前，请先了解 [DM 整体架构](/dm/dm-arch.md)以及各个关键组件的作用。
+
+## DM master 内部组成
+
+![DM high availability architecture](/media/dm/dm-ha-1.png)
+
+- **gRPC 和 HTTP 接口。**DM master 对外提供 gRPC 以及 HTTP 接口，供其他组件调用，如 dmctl、WebUI、DM worker。DM master 的 Follower 节点在收到 gRPC 和 HTTP 请求后，会 redirect 给 DM master 的 Leader 节点进行处理。
```
Comment (ran-huang) [inline dm/dm-high-availability.md:16]: 不太确定这篇文档面向的用户的 skill level......第一次提到 follower/leader 的时候，是否要给用户 link 到 raft 的相关文档？比如我们的[术语表](https://docs.pingcap.com/zh/tidb/stable/glossary#leaderfollowerlearner)

File: dm/dm-high-availability.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+从 DM master 的架构可以看出，DM master 的高可用主要依赖内嵌的 etcd 实现。
+
+若某一个 DM master 节点发生故障，DM master 的 Election 模块会通过 etcd 选举出新的 Leader 节点，新的 Leader 节点会启动 DM master 相关组件，且组件启动会读取保存于 etcd 中的信息，做到接续执行。
+
+### 其他组件恢复访问
+
+DM master 之外的组件（如 dmctl、DM worker）调用 DM master 接口，可采用传入多 endpoint 的方式，避免只配置单个 endpoint 时，该节点出现故障而无法访问的问题。
+
+其他组件（如 WebUI）若遇到访问节点挂掉，在 DM master 集群恢复后，可以切换节点继续访问。若非访问节点发生故障，DM master 集群恢复后可以继续访问。
```
Comment (ran-huang) [inline dm/dm-high-availability.md:49]: > DM master 之外的组件（如 dmctl、DM worker）
> 其他组件（如 WebUI）

这两处能否完全列举有哪些组件？现在的表述方式会让人疑惑第一段是否适用于 WebUI，第二段是否适用于 dmctl/dm worker

Comment (ran-huang) [reply dm/dm-high-availability.md:49]: “挂掉“有一点口语化，建议与上下文保持一致，统一用”发生故障“

File: dm/dm-high-availability.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+### 影响恢复的因素
+
+DM master 集群的 Leader 出现节点故障之后，会由 Election 模块重新选举一个 Leader，此过程主要受到 **etcd 性能**影响。
+
+新的 Leader 需要重新启动 Scheduler、Pessimist、Optimist 等模块，此过程主要需要读取保存于 etcd 中的各模块信息，实现接续执行。但此过程只是内存中实例构建，耗时不大，因此，耗时主要取决于 **etcd 集群的恢复时间**。
```
Comment (ran-huang) [inline dm/dm-high-availability.md:55]:
```suggestion
新的 Leader 需要重新启动 Scheduler、Pessimist、Optimist 等模块，此过程主要需要读取保存于 etcd 中的各模块信息，实现接续执行。但此过程只是在内存中构建实例，耗时不大，因此，耗时主要取决于 **etcd 集群的恢复时间**。
```

File: dm/dm-high-availability.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+当 DM worker 向 DM master 发送心跳中断超过 [TTL](/dm/dm-high-availability.md#dm-worker-与-dm-master-保持心跳的超时时间及配置) 后，DM master 会将该 DM worker 负责的 source 与对应 subtask 调度给没有绑定的 source 的 DM worker，调度的优先级从高到低为：
+
+1. 该 source 之前在该 DM worker 上存在未完成的全量导入任务。
+2. 该 source 上次由该 DM worker 处理且 DM worker 开启了该 source 的 relay log。
+3. 该 DM worker 开启了该 source 的 relay log。
+4. 该 source 上次由该 DM worker 处理。
+5. 任一空闲状态的 DM worker。
+
+如果没有处于空闲状态的 DM worker，则该 source 的 subtasks 将被暂时挂起，直到原先的 DM worker 恢复或者新的 DM worker 节点加入。当 DM master 收到 DM worker 的重新上线心跳或是新的 DM worker 节点的 heartbeat 信息时，DM master 会按照上述优先级顺序，寻找处于未绑定 DM worker 的 source 进行绑定。
```
Comment (ran-huang) [inline dm/dm-high-availability.md:84]:
```suggestion
如果没有处于空闲状态的 DM worker，则该 source 的 subtasks 将被暂时挂起，直到原先的 DM worker 恢复或者新的 DM worker 节点加入。当 DM master 收到 DM worker 的重新上线心跳或是新的 DM worker 节点的心跳信息时，DM master 会按照上述优先级顺序，寻找处于未绑定 DM worker 的 source 进行绑定。
```

File: dm/dm-high-availability.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+### 特殊任务调度
+
+通常情况下，DM master 只会为 source 试图绑定处于未绑定状态的 DM worker。但仅有一种情况存在例外，即 source 在某个 DM worker 存在未完成的全量导入任务。
+
+这样的调度策略是因为在 on-premises 环境中，DM worker 全量导出阶段导出的数据均保留在本地盘，大部分时候进行全量导入任务必须要在对应的 DM worker 才能正确进行，否则需要重新导出一遍数据，会增加大量不必要开销。
```
Comment (ran-huang) [inline dm/dm-high-availability.md:90]: on-premises 不算特别常见，能否翻译成中文？比如“本地部署”“私有部署”之类的，你来决定

File: dm/dm-high-availability.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    - 若不存在。worker2 绑定 source2 无变化，worker3 绑定 source1。
+
+DM worker 重新上线也与上述流程基本一致，如果寻找到符合条件的 source，会进行换绑并为 source 先前绑定的 DM worker 尝试重新绑定新的 source。
+
+### 当 DM worker 恢复后，source 是否会再迁回来
+
+在两种情况下 source 将再迁回：
+
+1. 如果 DM worker 之前连接的 source 在下线期间没有被调度到其他 DM worker （例如没有空闲 Worker）；
+2. 如果 source 之前在该 DM worker 上有未完成的 load 阶段任务，且 source 在其他 DM worker 上仍处于 dump 阶段。
```
Comment (ran-huang) [inline dm/dm-high-availability.md:111]:
```suggestion
在两种情况下 source 将再迁回恢复的旧 DM worker：

1. 旧 DM worker 之前连接的 source 在下线期间没有被调度到其他 DM worker （例如没有空闲 Worker）；
2. source 之前在旧 DM worker 上有未完成的 load 阶段任务，且 source 在新 DM worker 上仍处于 dump 阶段。
```

File: dm/dm-high-availability.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+在 DM v2.0.1 及之前版本，DM worker 的超时时间为 10 秒，在 DM v2.0.2 及更新版本中超时时间为 60s。该值可以通过修改 DM worker 的 `keepalive-ttl` 配置，单位为秒。
+
+在 DM v2.0.2 及更新版本中，DM worker 为 source 开启 relay-log 后，其超时时间将变为 1800 秒。该值可以通过修改 DM worker 的 `relay-keepalive-ttl` 配置，单位为秒。
+
+## DM 集群跨区域高可用
+
+由上文可知，DM master 的高可用依赖于 etcd 实现，而 etcd 高可用需要过半数节点存活。若仅考虑最大服务可用性，建议部署时选择奇数（≥3）个 Region/AZ 部署。DM master 每个 Region/AZ 至少部署 1 个以避免单个 Region/AZ 故障导致 DM 集群无法服务。
+
+由于单个 worker 仅能服务一个 source，因此每个 Region/AZ 还需要部署足够数量的空闲 worker，当某个 Region/AZ 故障时足以容纳该  Region/AZ 的所有 source。
```
Comment (ran-huang) [inline dm/dm-high-availability.md:127]:
```suggestion
由于单个 worker 仅能服务一个 source，因此每个 Region/AZ 还需要部署足够数量的空闲 worker，以便当某个 Region/AZ 故障时，空闲 worker 足以容纳该 Region/AZ 的所有 source。
```

File: dm/dm-glossary.md [hunk 10]
```diff
@@ -34,6 +34,10 @@ DM-worker 内部用于读取上游 Binlog 或本地 Relay log 并迁移到下游
 
 针对上游数据库实例表的黑白名单过滤功能，具体可参考 [Block & Allow Table Lists](/dm/dm-key-features.md#block--allow-table-lists)。该功能与 [MySQL Replication Filtering](https://dev.mysql.com/doc/refman/5.6/en/replication-rules.html) 及 [MariaDB Replication Filters](https://mariadb.com/kb/en/library/replication-filters/) 类似。
 
+### Bound
```
Comment (ran-huang) [inline dm/dm-glossary.md:37]: 添加了一些术语

File: dm/dm-high-availability.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+当 worker1 发生故障下线后，DM master 扫描所有 worker，并采取以下行为：
+
+- 判断 worker2 上是否存在 source1 未完成的全量迁移；
+    - 若存在。解绑 source2 ，worker2 绑定 source1 ，worker3 绑定 source2；
+    - 若不存在。worker2 绑定 source2 无变化，worker3 绑定 source1。
+
+DM worker 重新上线也与上述流程基本一致，如果寻找到符合条件的 source，会进行换绑，并为 source 先前绑定的 DM worker 尝试重新绑定新的 source。
+
+### 常见问题
```
Comment (ran-huang) [inline dm/dm-high-availability.md:106]: 这里添加了常见问题的 heading

File: dm/dm-glossary.md [hunk 12]
```diff
@@ -34,6 +34,10 @@ DM-worker 内部用于读取上游 Binlog 或本地 Relay log 并迁移到下游
...
+
+DM worker 与 source 的绑定关系，即一个 DM worker 在同一时间仅能处理一个 source 的迁移任务。当 DM worker 拉取某个 source 的 relay log 后，该 DM worker 将不能再处理其他 source 的迁移任务。
```
Comment (sunzhaoyang) [inline dm/dm-glossary.md:39]:
```suggestion
DM worker 与 source 的绑定关系，即一个 DM worker 在同一时间仅能处理一个 source 的迁移任务。当 DM worker 开始接收某个 source 的 binlog 后，该 DM worker 将不能再处理其他 source 的迁移任务。
```

File: dm/dm-high-availability.md (hunk 3 above)
Comment (sunzhaoyang) [reply dm/dm-high-availability.md:16]: 没事，了解 TiDB 的应该知道一点

File: dm/dm-high-availability.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+
+- **etcd。**DM master 内部嵌入 etcd 来组成集群，DM master 的 gRPC 和 HTTP 接口 attach 到 etcd 中，同时 etcd 也作为 DM master、DM worker、Source、Task 等配置或者状态的存储介质。
+
+- **Election。**Election 周期性地调用 etcd 的 campaign 接口进行选举。
+
+    - 若该节点为 Leader，则启动 DM master 的其他组件，如 Scheduler、Pessimist、Optimist 等，各组件在启动时会读取 etcd 中保存的信息，实现相关任务的接续执行。
+    - 若该节点为 Follower，则不启动其他组件。
+    - 若该节点从 Leader 切换回 Follower，则会关闭其他组件。
+
+- **Scheduler。**负责注册及监听 DM worker 状态，安排分配 Source 以及 Task，可参考 DM worker 任务分配的相关文档进一步了解。
```
[the remark this answers is not in the document]
Comment (sunzhaoyang) [reply dm/dm-high-availability.md:26]: 6.2 发版之后会加一个

File: dm/dm-high-availability.md [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+
+从 DM master 的架构可以看出，DM master 的高可用主要依赖内嵌的 etcd 实现。
+
+若某一个 DM master 节点发生故障，DM master 的 Election 模块会通过 etcd 选举出新的 Leader 节点，新的 Leader 节点会启动 DM master 相关组件，且组件启动会读取保存于 etcd 中的信息，做到接续执行。
+
+### 其他组件恢复访问
+
+DM master 之外的组件（如 dmctl、DM worker）调用 DM master 接口，可采用传入多 endpoint 的方式，避免只配置单个 endpoint 时，该节点出现故障而无法访问的问题。
+
+其他组件（如 WebUI）若遇到访问节点挂掉，在 DM master 集群恢复后，可以切换节点继续访问。若非访问节点发生故障，DM master 集群恢复后可以继续访问。
```
Comment (sunzhaoyang) [inline dm/dm-high-availability.md:49]:
```suggestion
其他组件（当前指 WebUI）若遇到访问节点不可用，在 DM master 集群恢复后，可以切换节点继续访问。若非访问节点发生故障，DM master 集群恢复后可以继续访问。
```

**Review by ran-huang — APPROVED** [review]

File: dm/dm-high-availability.md [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+
+在详细了解 DM 高可用机制之前，请先了解 [DM 整体架构](/dm/dm-arch.md)以及各个关键组件的作用。
+
+## DM master 内部架构
+
+![DM master 高可用架构](/media/dm/dm-high-availability-1.png)
+
+- **gRPC 和 HTTP 接口。**DM master 对外提供 gRPC 以及 HTTP 接口，供其他组件调用，如 dmctl、WebUI、DM worker。DM master 的 Follower 节点在收到 gRPC 和 HTTP 请求后，会 redirect 给 DM master 的 Leader 节点进行处理。
+
+- **etcd。**DM master 内部嵌入 etcd 来组成集群，DM master 的 gRPC 和 HTTP 接口 attach 到 etcd 中，同时 etcd 也作为 DM master、DM worker、Source、Task 等配置或者状态的存储介质。
```
Comment (lichunzhu) [inline dm/dm-high-availability.md:18]: > DM master 的 gRPC 和 HTTP 接口 attach 到 etcd 中

这句是什么意思？

File: dm/dm-high-availability.md [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+- **Scheduler。**负责注册及监听 DM worker 状态，安排分配 Source 以及 Task，可参考 DM worker 任务分配的相关文档进一步了解。
+
+- **Pessimist 和 Optimist。**DM master 实现对 DDL 进行悲观协调以及乐观协调的模块。
+
+- **dmctl/WebUI 调用流程。**dmctl 调用 DM master 的 gRPC 接口，WebUI 调用 DM master 的 HTTP 接口。
+
+    - 若非 Leader 节点，则进行 redirect 给 Leader 节点。
+    - 若 dmctl 和 WebUI 请求需要调用 DM worker 进行信息收集或者操作，则由 DM master 的 Scheduler 模块通过 gRPC 接口调用 DM worker。
+
+- **DM master 与 DM worker 调用流程。**DM worker 启动时会通过 gRPC 调用 DM master 来注册 worker 信息，DM master 主要调用 Scheduler 模块实现相关的逻辑。其中，Scheduler 会将 DM worker 信息写入到 etcd，并对其进行监听。DM worker 在注册之后，会通过 etcd 实现 Keep-Alive。若 DM worker 发生故障，DM master 的 Scheduler 模块的监听会发现该 DM worker 出现故障，从而将该 DM worker 上的相关任务进行转移。
```
Comment (lichunzhu) [inline dm/dm-high-availability.md:35]: Scheduler 不写 worker 信息，或者说改成写调度信息？
```suggestion
- **DM master 与 DM worker 调用流程。**DM worker 启动时会通过 gRPC 调用 DM master 来注册 worker 信息，DM master 主要调用 Scheduler 模块实现相关的逻辑。DM worker 在注册之后，会通过 etcd 实现 Keep-Alive。若 DM worker 发生故障，DM master 的 Scheduler 模块的监听会发现该 DM worker 出现故障，从而将该 DM worker 上的相关任务进行转移。
```

File: dm/dm-high-availability.md (hunk 15 above)
Comment (WizardXiao) [reply dm/dm-high-availability.md:18]: 其实是想表达这里通过 etcd api 实现 grpc 和 http 的意思，不过表达出来感觉不是很贴切。。可能 `服务注册以及服务发现` 会好一些吧？
https://github.com/pingcap/tiflow/blob/45c837b97e3a9fc47b05bfd4e7c7aa10db956012/dm/dm/master/etcd.go#L46

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-6.6 label because the v6.6 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.6 and will no longer receive new updates.

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-7.0 label because the v7.0 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v7.0 and will no longer receive new updates.

**Comment by qiancai** [comment]
Removed the `needs-cherry-pick-release-7.2` label because the v7.2 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v7.2 and will no longer receive new updates.

File: dm/dm-high-availability.md (hunk 15 above)
Comment (lilin90) [inline dm/dm-high-availability.md:18]: 根据 comment 讨论进行的修改。
```suggestion
- **etcd。**DM master 内部嵌入 etcd 来组成集群，同时 etcd 也作为 DM master、DM worker、Source、Task 等配置或者状态的存储介质。
```

File: dm/dm-high-availability.md [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+
+- **etcd。**DM master 内部嵌入 etcd 来组成集群，DM master 的 gRPC 和 HTTP 接口 attach 到 etcd 中，同时 etcd 也作为 DM master、DM worker、Source、Task 等配置或者状态的存储介质。
+
+- **Election。**Election 周期性地调用 etcd 的 campaign 接口进行选举。
+
+    - 若该节点为 Leader，则启动 DM master 的其他组件，如 Scheduler、Pessimist、Optimist 等，各组件在启动时会读取 etcd 中保存的信息，实现相关任务的接续执行。
+    - 若该节点为 Follower，则不启动其他组件。
+    - 若该节点从 Leader 切换回 Follower，则会关闭其他组件。
+
+- **Scheduler。**负责注册及监听 DM worker 状态，安排分配 Source 以及 Task，可参考 DM worker 任务分配的相关文档进一步了解。
```
Comment (lilin90) [inline dm/dm-high-availability.md:26]:
```suggestion
- **Scheduler。**负责注册及监听 DM worker 状态，安排分配 Source 以及 Task。
```

**Review by Frank945946 — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

File: dm/dm-high-availability.md [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+
+- worker1 绑定 source1，subtask 处于 dump 阶段；
+- worker2 绑定 source2，subtask 处于 sync 阶段；
+- worker3 空闲；
+
+当 worker1 发生故障下线后，DM master 扫描所有 worker，并采取以下行为：
+
+- 判断 worker2 上是否存在 source1 未完成的全量迁移；
+    - 若存在。解绑 source2 ，worker2 绑定 source1 ，worker3 绑定 source2；
+    - 若不存在。worker2 绑定 source2 无变化，worker3 绑定 source1。
```
Comment (hfxsd) [inline dm/dm-high-availability.md:102]:
```suggestion
当 worker1 发生故障下线后，DM master 扫描所有 worker，并判断 worker2 上是否存在 source1 未完成的全量迁移；

- 若存在。解绑 source2 ，worker2 绑定 source1，worker3 绑定 source2；
- 若不存在。worker2 绑定 source2 无变化，worker3 绑定 source1。
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**